doc/dev/dataplane_codegen_quick_start.md

```diff
@@ -0,0 +1,198 @@
+Usually this name comes from the name of the REST API file itself or one of the directories toward the end of the file path.
+
+In Python, a project's `package name` will normally be `azure-{service_name}-{module_name}`. After release, you can find it in pypi.
+For example: you can find [azure-messaging-webpubsubservice](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/webpubsub/azure-messaging-webpubsubservice) in [pypi](https://pypi.org/project/azure-messaging-webpubsubservice/).
+
+## Project folder structure
+
+Before we start, we probably should get to know the project folder for [SDK repo](https://github.com/Azure/azure-sdk-for-python).
+
+All the files for your SDK will reside in the **${PROJECT_ROOT} folder**, where
+
+```sh
+${PROJECT_ROOT}=sdk/{service_name}/azure-{service_name}-{module_name}
+```
+
+Normally, the folder structure would be something like:
+
+- **${PROJECT_ROOT}**
+  - `azure/{service_name}/{module_name}` : folder where generated code is.
+  - `swagger`: folder of configuration file which is used to generate SDK code with autorest
+  - `tests`: folder of test files
+  - `samples`: folder of sample files
+
+More details on the structure of Azure SDK repos is available in the [Azure SDK common repo](https://github.com/Azure/azure-sdk/blob/main/docs/policies/repostructure.md#sdk-directory-layout).
+
+## How to generate SDK code with Dataplane Codegen
+
+We are working on to automatically generate everything right now, but currently we still need some manual work to get a releasable package. Here're the steps of how to get the package.
+
+1. **Create the ${PROJECT_ROOT} folder if it does not already exist**
+
+2. **Create a README.md file under ${PROJECT_ROOT}/swagger**
+
+   This README.md file will hold the options to be passed to autorest to generate the code. Storing these options in a file in the
+   SDK repo helps make SDK generation reproducible.
+
+   ````markdown
+   # Azure Purview for Python
+  
+   > see https://aka.ms/autorest
+  
+   ### Settings
+  
+   ```yaml
+   input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/webpubsub/data-plane/WebPubSub/stable/2021-10-01/webpubsub.json
+   output-folder: ../azure/messaging/webpubsubservice
+   namespace: azure.messaging.webpubsubservice
+   package-name: azure-messaging-webpubsubservice
+   license-header: MICROSOFT_MIT_NO_VERSION
+   clear-output-folder: true
+   no-namespace-folders: true
+   python: true
+   title: WebPubSubServiceClient
+   version-tolerant: true
+   package-version: 1.0.0b1
+   add-credential: true
+   credential-scopes: https://webpubsub.azure.com/.default
+   ```
+   ````
+
+   **Reviewers: which of the above can we remove?  We should keep only those needed for the "common" case. And if any could just be the default (e.g. license-header), we should remove those too. Then we should document whatever remains.**
```

Comment:
The title should be extracted from the Swagger titlte, so title should be used only if the Swagger titlte is not the expected client titlte for some reason.

We have our ongoing discussion on credentials (that would be two flags less).

I don't think we can remove anything else  at the momemt, even the licence header I think is not default to MICROSOFT_MIT_NO_VERSION....

We _could_ make a rule to deduce the default namespace from the package name (cc @iscai-msft).

Reply:
yeah i think that's fair, made an [issue](https://github.com/Azure/autorest.python/issues/1121) for it